File: frontend/src/types/appointment.ts
export type AppointmentStatus = "scheduled" | "confirmed" | "completed" | "cancelled" | "no_show";
export type Appointment = {
  id: number; patient_id: number; doctor_id: number; center_id: number | null;
  appointment_date: string; appointment_time: string;
  reason: string | null; status: AppointmentStatus; notes: string | null;
  patient_name: string; doctor_name: string; center_name: string | null; center_city: string | null;
  created_at: string; updated_at: string;
};
export type AppointmentInput = {
  patient_id: number;
  doctor_id: number | null;
  center_id: number | null;
  appointment_date: string;
  appointment_time: string;
  reason: string | null;
  status: AppointmentStatus;
  notes: string | null;
};

File: backend/app/schemas/appointment.py
from datetime import date, time, datetime
from pydantic import BaseModel, Field

class AppointmentBase(BaseModel):
    patient_id: int
    appointment_date: date
    appointment_time: time
    reason: str | None = Field(default=None, max_length=5000)
    status: str = Field(default="scheduled", pattern="^(scheduled|confirmed|completed|cancelled|no_show)$")
    notes: str | None = Field(default=None, max_length=10000)

class AppointmentCreate(AppointmentBase):
    pass

class AppointmentResponse(AppointmentBase):
    id: int
    doctor_id: int
    patient_name: str
    doctor_name: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
